src/cubi_tk/snappy/pull_sheets.py

```diff
@@ -173,7 +173,21 @@ def build_sheet(
         sodar_api_token=config.global_config.sodar_api_token,
         project_uuid=project_uuid,
     )
-    isa = isa_dict_to_isa_data(isa_dict, assay_txt)
+    assay_filename = None
+    if(assay_uuid): #samplesheet.export doesnt pull assayuuids, get assauuuid via samplesheet.retrive
+        investigation = api.samplesheet.retrieve(
+            sodar_url=config.global_config.sodar_server_url,
+            sodar_api_token=config.global_config.sodar_api_token,
+            project_uuid=project_uuid,
+        )
+        assay = None
+        for study in investigation.studies.values():
+            for remote_assay_uuid in study.assays.keys():
+                if assay_uuid== remote_assay_uuid:
+                    assay = study.assays[remote_assay_uuid]
+                    assay_filename = assay.file_name
+                    break
+    isa = isa_dict_to_isa_data(isa_dict, assay_filename)
```

Comment:
If the given assay UUID doesn't exist/match anything, then from what I see cubi-tk will simply (and I think silently) load all assays here.
There sould be at least a warning, maybe bettern an error if the given assay_uuid doesn't match (one of multiple UUIDs)

Reply:
Additionally (somewhat adressing the multiple project issue): if only one assay is found for agiven project/investigation, then maybe that assay should always be used, regardless of assay_uuid definition? (or maybe give a warning but ignore assay_uuid)